### PR TITLE
Add HereDoc support

### LINKED ISSUE #1: Dockerfile.prod and it's varieties are not getting detected.
I have attached a screenshot of the issue: 

CleanShot 2024-02-23 at 11 32 36

### LINKED ISSUE #5: Heredoc support absent
While using [heredocs](https://collabnix.com/using-heredocs-in-dockerfiles-simplify-your-image-building-process/) with Dockerfile, the syntax is broken from heredoc invocation till the end of the file.

### PR DESCRIPTION
**Pull request by joshmeads**
Update tree-sitter-dockerfile to add support for heredocs(camdencheek/tree-sitter-dockerfile#45)

Close #1
Fixes #5
Fixes zed-industries/zed#8577

### CODE REVIEW

File: extension.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 id = "dockerfile"
 name = "Dockerfile"
 description = "Dockerfile support."
-version = "0.0.3"
+version = "0.0.4"
 schema_version = 1
 authors = ["d1y <chenhonzho@gmail.com>"]
```
Comment (d1y) [inline extension.toml:6]: Can you also fill yourself in as an author?

**Comment by d1y** [comment]
Can you help me update README.md by the way?

```diff
diff --git a/.gitignore b/.gitignore
index 3d48c45..5c2bfcd 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,2 +1,3 @@
 target/
+grammars
 *.wasm
diff --git a/README.md b/README.md
index 46cc42c..0fcf58d 100644
--- a/README.md
+++ b/README.md
@@ -1,6 +1,16 @@
-Prev PR: https://github.com/zed-industries/zed/pull/7977 & https://github.com/zed-industries/zed/issues/6905
-
 # Dockerfile
 
 - Tree Sitter: [tree-sitter-dockerfile](https://github.com/camdencheek/tree-sitter-dockerfile)
 - Language Server: [dockerfile-language-server](https://github.com/rcjsuen/dockerfile-language-server)
+
+Now zed support match [glob](https://github.com/zed-industries/zed/pull/12043). ex:
+
+```json
+{
+  "file_types": {
+    "Dockerfile": [ "Dockerfile.*" ]
+  }
+}
+```
+
+Prev PR: https://github.com/zed-industries/zed/pull/7977 & https://github.com/zed-industries/zed/issues/6905

```

**Comment by joshmeads** [comment]
Sure, I marked it as draft again as I'm having issues testing it locally.

Zed isn't picking it up when I click install dev extension. I'll continue testing tomorrow. First time messing with Zed extensions and I'm not a rust dev so trying to figure this out by trial and error :). 

Thanks for making this btw!

**Comment by d1y** [comment]
> Sure, I marked it as draft again as I'm having issues testing it locally.
> 
> Zed isn't picking it up when I click install dev extension. I'll continue testing tomorrow. First time messing with Zed extensions and I'm not a rust dev so trying to figure this out by trial and error :).
> 
> Thanks for making this btw!

May I ask what problem you encountered while debugging? Based on my experience

1. You need install wasm32-wasi dep: `rustup target add wasm32-wasi`
2. Please confirm if two trusts have been installed? Did rustup and brew install it? please uninstall a tool such as `brew uninstall rust`(👆🏻 1 continuing with)
3. Please remove the local dockerfile extension first before installing the development extension
4. View debugging using `zed: open log`

**Comment by joshmeads** [comment]
@d1y thank you! you were correct about there being two rusts installed. I always use rustup but I guess zed was detecting a brew version! Uninstalling it resolved the issue.

**Comment by d1y** [comment]
LTGM 🎉

**Pull request merged**